parameters/parser/ParamsParser.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	if len(tag) != TagLen {
+		return nil, fmt.Errorf("invalid tag length, expected %d, got %d", TagLen, len(tag))
+	}
+
+	if len(p.CovenantPks) == 0 {
+		return nil, fmt.Errorf("empty covenant public keys")
+	}
+	if p.CovenantQuorum > uint64(len(p.CovenantPks)) {
+		return nil, fmt.Errorf("covenant quorum cannot be more than the amount of covenants")
+	}
+
+	quroum, err := parseUint32(p.CovenantQuorum)
+	if err != nil {
+		return nil, fmt.Errorf("invalid covenant quorum: %w", err)
+	}
+
+	var covenantKeys []*btcec.PublicKey
+	for _, covPk := range p.CovenantPks {
+		pk, err := parseCovenantPubKeyFromHex(covPk)
+		if err != nil {
+			return nil, fmt.Errorf("invalid covenant public key %s: %w", covPk, err)
+		}
+
+		covenantKeys = append(covenantKeys, pk)
+	}
+
+	maxStakingAmount, err := parseBtcValue(p.MaxStakingAmount)
+
+	if err != nil {
+		return nil, fmt.Errorf("invalid max_staking_amount: %w", err)
+	}
+
+	minStakingAmount, err := parseBtcValue(p.MinStakingAmount)
+
+	if err != nil {
+		return nil, fmt.Errorf("invalid min_staking_amount: %w", err)
+	}
+
+	if maxStakingAmount <= minStakingAmount {
+		return nil, fmt.Errorf("max-staking-amount must be larger than min-staking-amount")
+	}
+
+	ubTime, err := parseTimeLockValue(p.UnbondingTime)
+	if err != nil {
+		return nil, fmt.Errorf("invalid unbonding_time: %w", err)
+	}
+
+	ubFee, err := parseBtcValue(p.UnbondingFee)
+	if err != nil {
+		return nil, fmt.Errorf("invalid unbonding_fee: %w", err)
+	}
+
+	maxStakingTime, err := parseTimeLockValue(p.MaxStakingTime)
+	if err != nil {
+		return nil, fmt.Errorf("invalid max_staking_time: %w", err)
+	}
+
+	minStakingTime, err := parseTimeLockValue(p.MinStakingTime)
+	if err != nil {
+		return nil, fmt.Errorf("invalid min_staking_time: %w", err)
+	}
+
+	// NOTE: Allow config when max-staking-time is equal to min-staking-time, as then
+	// we can configure a fixed staking time.
+	if maxStakingTime < minStakingTime {
+		return nil, fmt.Errorf("max-staking-time must be larger or equalt min-staking-time")
+	}
+
+	confirmationDepth, err := parseConfirmationDepthValue(p.ConfirmationDepth)
+	if err != nil {
+		return nil, fmt.Errorf("invalid confirmation_depth: %w", err)
+	}
+
+	stakingCap, err := parseBtcValue(p.StakingCap)
+	if err != nil {
+		return nil, fmt.Errorf("invalid staking_cap: %w", err)
+	}
+
+	return &ParsedVersionedGlobalParams{
+		Version:           p.Version,
+		ActivationHeight:  p.ActivationHeight,
+		StakingCap:        stakingCap,
+		Tag:               tag,
+		CovenantPks:       covenantKeys,
+		CovenantQuorum:    quroum,
+		UnbondingTime:     ubTime,
+		UnbondingFee:      ubFee,
+		MaxStakingAmount:  maxStakingAmount,
+		MinStakingAmount:  minStakingAmount,
+		MaxStakingTime:    maxStakingTime,
+		MinStakingTime:    minStakingTime,
+		ConfirmationDepth: confirmationDepth,
+	}, nil
+}
+
+func (g *ParsedGlobalParams) GetVersionedGlobalParamsByHeight(btcHeight uint64) *ParsedVersionedGlobalParams {
```

Comment:
Shall we add a note that `nil` will be returned if the `btcHeight` is lower than the earliest activation height? Or would it be clearer if we return err if no available params found so that we force the caller to handler errors?

Reply:
hmh this is a bit philosophical issue as returning `nil` means essentially that there is not params for given height, so `nil` is used as to convey information about optionality of result. 

On the other hand, this can also be seen as error which convey  information roughly meaning `provided height does bit have sense in context of this instance if parsed params`

Both approaches makes sense imo.

As tie breaker, it seems 3 of our services return `nil` (staking api service, pipeline, signer) and one returns error  (indexer), therefore I will just add documentation here (as this will lead to less breakage across services when converting to using library)

Reply:
Either way makes sense to me